benchmarks/event/src/main.rs

```diff
@@ -39,50 +40,49 @@ fn perform_benchmark<T: Service>(args: &Args) {
         .create()
         .unwrap();
 
-    let barrier = Barrier::new(3);
+    let barrier_handle = BarrierHandle::new();
+    let barrier = BarrierBuilder::new(3).create(&barrier_handle).unwrap();
+
+    let t1 = ThreadBuilder::new().affinity(0).priority(255).spawn(|| {
+        let notifier_a2b = service_a2b.notifier_builder().create().unwrap();
+        let listener_b2a = service_b2a.listener_builder().create().unwrap();
 
-    std::thread::scope(|s| {
-        let t1 = s.spawn(|| {
-            let notifier_a2b = service_a2b.notifier().create().unwrap();
-            let listener_b2a = service_b2a.listener().create().unwrap();
+        barrier.wait();
+        notifier_a2b.notify().expect("failed to notify");
 
-            barrier.wait();
+        for _ in 0..args.iterations {
+            while listener_b2a.blocking_wait_one().unwrap().is_none() {}
             notifier_a2b.notify().expect("failed to notify");
+        }
+    });
 
-            for _ in 0..args.iterations {
-                while listener_b2a.blocking_wait_one().unwrap().is_none() {}
-                notifier_a2b.notify().expect("failed to notify");
-            }
-        });
-
-        let t2 = s.spawn(|| {
-            let notifier_b2a = service_b2a.notifier().create().unwrap();
-            let listener_a2b = service_a2b.listener().create().unwrap();
-
-            barrier.wait();
-            for _ in 0..args.iterations {
-                while listener_a2b.blocking_wait_one().unwrap().is_none() {}
-                notifier_b2a.notify().expect("failed to notify");
-            }
-        });
-
-        std::thread::sleep(std::time::Duration::from_millis(100));
-        let start = Instant::now();
-        barrier.wait();
+    let t2 = ThreadBuilder::new().affinity(1).priority(255).spawn(|| {
+        let notifier_b2a = service_b2a.notifier_builder().create().unwrap();
+        let listener_a2b = service_a2b.listener_builder().create().unwrap();
 
-        t1.join().expect("thread failure");
-        t2.join().expect("thread failure");
-
-        let stop = start.elapsed();
-        println!(
-            "{} ::: MaxEventId: {}, Iterations: {}, Time: {}, Latency: {} ns",
-            std::any::type_name::<T>(),
-            args.max_event_id,
-            args.iterations,
-            stop.as_secs_f64(),
-            stop.as_nanos() / (args.iterations as u128 * 2)
-        );
+        barrier.wait();
+        for _ in 0..args.iterations {
+            while listener_a2b.blocking_wait_one().unwrap().is_none() {}
+            notifier_b2a.notify().expect("failed to notify");
+        }
     });
+
+    std::thread::sleep(std::time::Duration::from_millis(100));
+    let start = Time::now().expect("failed to acquire time");
+    barrier.wait();
```

Comment:
Why not just
```suggestion
    barrier.wait();
    let start = Time::now().expect("failed to acquire time");
```

Reply:
We start the time measurements before wait, to be a bit more conservative and maybe measure a bit more than we have to.

If I remove the sleep on certain OSes like Windows, the immense thread-creation time influences the benchmark more negatively, so I give it additional 100ms to ensure that everything is already ready to use and then we go into the barrier.

Reply:
You could use a barrier-timestamp-barrier sandwich to not rely too much on timing